shared/src/commonMain/kotlin/co/touchlab/kampstarter/models/BaseModel.kt

```diff
@@ -1,32 +1,38 @@
 package co.touchlab.kampstarter.models
 
 import co.touchlab.kampstarter.printThrowable
+import co.touchlab.kermit.Kermit
 import kotlinx.coroutines.CoroutineExceptionHandler
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.SupervisorJob
 import org.koin.core.KoinComponent
+import org.koin.core.inject
+import org.koin.core.parameter.parametersOf
 import kotlin.coroutines.CoroutineContext
 
 open class BaseModel : KoinComponent {
-    internal val scope = MainScope(Dispatchers.Main)
+    private val log: Kermit by inject { parametersOf("BaseModel") }
```

Comment:
We're using constructor injection most other spots. Any reason to do field injection in the models?

Reply:
Mainly legacy and could be better. The BreedModel is the boundary between shared and platform, so we didn't want to pull concerns out into the platform that we didn't have to. We could move BreedModel instantiation to Koin, but the syntax for fetching it and passing the callbacks it needs is going to be messy, particularly on iOS. The helper I added in KoiniOS currently only takes one parameter, would need more work to support this.

Reply:
Ok. I'm fine with this for now but this is probably worth thinking deeper about down the road. I'm thinking maybe `KoinIOS` becomes a somewhat more general `IosAdapter` sort of container, like the repository wrapper we used on GB.